simulation/simulation.py

```diff
@@ -0,0 +1,210 @@
+from importlib.resources import path
+from data.data_request import Token_Pair
+import QuantLib as ql
+import numpy as np
+from datetime import datetime
+import pandas as pd
+
+
+def parse_date_to_quantlib(date: pd.Timestamp) -> ql.Date:
+    """Parses a pandas.Timestamp object into a quantLib.Date object.
+
+    Args:
+        date (pd.Timestamp): Timestamp object to parse.
+
+    Returns:
+        ql.Date: quantLib.Date object for the simulation.
+    """
+    date = datetime.strftime(date, "%d-%m-%Y")
+    return ql.Date(*[int(i) for i in date.split("-")])
+
+
+# TODO: Can this be generalized so that all processes can use this function for creating a path generator?
+def path_generator(process, maturity: float, nSteps: int):
+    """Returns a path generator for any process by first, generating a uniform_sequence_generator for the given dimensions.
+    Secondly, it creates a gaussian_sequence_generator, using the uniform_sequence_generator. Lastly it returns a GaussianMultiPathGenerator iterator-object,
+    which can be used to generate random paths using the given process.
+
+    Args:
+        process (_type_): _description_
+        maturity (_type_): The maruity at which the process ends. This can be though of as how many times the number of steps will be run by the simulation.
+            E.g. if you have steps that represent one day, you can create a time series with the maturity of two years by using 365 steps and a maturity of 2.
+        nSteps (_type_): The number of steps within one period of the maturity. The unit is determined by the time units of the sample. 
+            E.g. if you have estimate the mean and standard deviation from a sample of daily data, each step will be generated based on those values.
+
+    Returns:
+        _type_: _description_
+    """
+    dimensions = process.factors()
+    times = ql.TimeGrid(maturity, nSteps)
+
+    unifrom_sequence_generator = ql.UniformRandomSequenceGenerator(
+        dimensions * nSteps, ql.UniformRandomGenerator()
+    )
+    gaussian_sequence_generator = ql.GaussianRandomSequenceGenerator(
+        unifrom_sequence_generator
+    )
+    return ql.GaussianMultiPathGenerator(
+        process, list(times), gaussian_sequence_generator, False
+    )
+
+
+class Simulation:
+    """This class represents a simulation with different processes.
+    """
+
+    def __init__(self, token_pair: Token_Pair, strategy: str) -> None:
+        self._token_pair = token_pair
+        self._strategy = strategy
```

Comment:
These properties alone are not enough to warrant the usage of a class, I think just having methods that take these as parameters.

The quantlib wrappers should be in their own [module](https://docs.python.org/3/tutorial/modules.html), while `simulate` just calles them from this module. In other parts of the code too, as mentioned, it's cleaner to use a module instead of a class.